### PR TITLE
Refined the interface to use gfx::Stream and better factory methods

### PR DESCRIPTION
**Pull request by kvark**
This is a partially breaking change!

### CODE REVIEW

File: src/text_renderer.rs [hunk 1]
```diff
@@ -42,9 +41,7 @@ impl<R: gfx::Resources> TextRenderer<R> {
             .. gfx::ShaderSource::empty()
         };
 
-        let program = match factory.link_program_source(
-            vertex, fragment, &device_capabilities
-        ) {
+        let program = match factory.link_program_source2(vertex, fragment) {
```
Comment (stjahns) [inline src/text_renderer.rs:44]: nice haha

**Comment by stjahns** [comment]
Looks good to me! Thanks! I should finally have a little free time to unbreak some things tomorrow, sorry for being out of touch the last few days.

**Comment by Kagami** [comment]
Hm, how do I pass `&mut factory` and `canvas: &mut Stream<R>` at the same time if I only have canvas instance from `gfx_window_glutin`?

**Comment by kvark** [comment]
@Kagami this PR is doing exactly that:

``` rust
let mut stream = (&mut canvas.renderer, &canvas.output);
elf.render(&mut stream, &mut canvas.factory, projection);
```

**Comment by Kagami** [comment]
Ah, I see, Stream is implemented for tuple of renderer and output. Thanks.

But I can't use `canvas` after that... Required to wrap it into separate block.

**Pull request merged**